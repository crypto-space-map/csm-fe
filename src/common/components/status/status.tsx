import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { COLOR_PALLETTE } from 'global/pallette';
import { gradientBackground, gradientBorder } from 'global/styles';

export type StatusProps = React.HtmlHTMLAttributes<HTMLDivElement> & {
  size?: 's' | 'm' | 'l';
  variant?: 'text' | 'outlined' | 'contained' | 'black';
  text: string;
};

const getStatusVariant = (variant: StatusProps['variant']) => {
  switch (variant) {
    case 'outlined':
      return css`
        background: ${COLOR_PALLETTE.MAIN_WHITE};
        ${gradientBorder({ borderRadius: 20 })}
      `;
    case 'black':
      return css`
        background: ${COLOR_PALLETTE.MAIN_LAYOUT};
        color: ${COLOR_PALLETTE.MAIN_WHITE};
      `;
    case 'text':
      return css`
        background: none;
        color: ${COLOR_PALLETTE.MAIN_WHITE};
      `;
    case 'contained':
    default:
      return css`
        ${gradientBackground}
      `;
  }
};

const getStatusSize = (size: StatusProps['size']) => {
  switch (size) {
    default:
    case 'm':
      return css`
        padding: 4px 8px;
        font-size: 12px;
      `;
    case 'l':
      return css`
        padding: 8px 16px;
        font-size: 16px;
        line-height: 22px;
      `;
  }
};

const StyledStatus = styled.div<StatusProps>`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #1d1c1c;
  border-radius: 20px;
  text-align: center;
  ${({ variant }) => getStatusVariant(variant)}
  ${({ size }) => getStatusSize(size)}
`;

export const Status = (props: StatusProps) => <StyledStatus {...props}>{props.text}</StyledStatus>;
