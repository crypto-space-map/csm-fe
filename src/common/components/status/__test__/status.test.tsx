import { render } from '@testing-library/react';

import { StatusProps } from '..';
import { Status } from '../status';

const props: StatusProps[] = [
  {
    size: 'l',
    variant: 'contained',
    text: 'contained',
  },
  {
    size: 'm',
    variant: 'outlined',
    text: 'outlined',
  },
  {
    size: 's',
    variant: 'text',
    text: 'text',
  },
];

describe('<Status />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <>
        {props.map(item => (
          <Status {...item} key={item.variant} />
        ))}
      </>
    );
    expect(container).toMatchSnapshot();
  });
  it('should be defined', () => {
    const { container } = render(
      <>
        {props.map(item => (
          <Status {...item} key={item.variant} />
        ))}
      </>
    );
    expect(container).toBeDefined();
  });
});
