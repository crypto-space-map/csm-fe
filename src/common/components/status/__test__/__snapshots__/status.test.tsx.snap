// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should match snapshot 1`] = `
<div>
  <div
    class="css-focol9"
  >
    contained
  </div>
  <div
    class="css-qs09xa"
  >
    outlined
  </div>
  <div
    class="css-q5ymd1"
  >
    text
  </div>
</div>
`;
